<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: Single Cell Overview of Normalized Expression data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>
<h1> Single Cell Overview of Normalized Expression data
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;scone&rsquo; version 1.3.0</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<table width="100%">
<tr><td style="width: 25%;"><a href="biplot_color.html">biplot_color</a></td>
<td>Function for biplotting with no point labels and with points color-coded according to a quantitative variable. For example: the rank of normalization performance.</td></tr>
<tr><td style="width: 25%;"><a href="biplot_interactive.html">biplot_interactive</a></td>
<td>Interactive biplot</td></tr>
<tr><td style="width: 25%;"><a href="control_genes.html">cellcycle_genes</a></td>
<td>Data: Positive and Negative Control Genes</td></tr>
<tr><td style="width: 25%;"><a href="CLR_FN.html">CLR_FN</a></td>
<td>Centered log-ratio (CLR) normalization wrapper function</td></tr>
<tr><td style="width: 25%;"><a href="control_genes.html">control_genes</a></td>
<td>Data: Positive and Negative Control Genes</td></tr>
<tr><td style="width: 25%;"><a href="control_genes.html">cortical_markers</a></td>
<td>Data: Positive and Negative Control Genes</td></tr>
<tr><td style="width: 25%;"><a href="DESEQ_FN.html">DESEQ_FN</a></td>
<td>Relative log-expression (RLE; DESeq) scaling normalization wrapper function</td></tr>
<tr><td style="width: 25%;"><a href="estimate_ziber.html">estimate_ziber</a></td>
<td>Parameter estimation of zero-inflated bernoulli model</td></tr>
<tr><td style="width: 25%;"><a href="factor_sample_filter.html">factor_sample_filter</a></td>
<td>Factor-based Sample Filtering: Function to filter single-cell RNA-Seq libraries.</td></tr>
<tr><td style="width: 25%;"><a href="fast_estimate_ziber.html">fast_estimate_ziber</a></td>
<td>Fast parameter estimation of zero-inflated bernoulli model</td></tr>
<tr><td style="width: 25%;"><a href="FQ_FN.html">FQT_FN</a></td>
<td>Full-quantile normalization wrapper function</td></tr>
<tr><td style="width: 25%;"><a href="FQ_FN.html">FQ_FN</a></td>
<td>Full-quantile normalization wrapper function</td></tr>
<tr><td style="width: 25%;"><a href="get_bio.html">get_batch</a></td>
<td>Get Factor of Biological Conditions and Batch</td></tr>
<tr><td style="width: 25%;"><a href="get_bio.html">get_batch-method</a></td>
<td>Get Factor of Biological Conditions and Batch</td></tr>
<tr><td style="width: 25%;"><a href="get_bio.html">get_bio</a></td>
<td>Get Factor of Biological Conditions and Batch</td></tr>
<tr><td style="width: 25%;"><a href="get_bio.html">get_bio-method</a></td>
<td>Get Factor of Biological Conditions and Batch</td></tr>
<tr><td style="width: 25%;"><a href="get_design.html">get_design</a></td>
<td>Retrieve Design Matrix</td></tr>
<tr><td style="width: 25%;"><a href="get_design.html">get_design-method</a></td>
<td>Retrieve Design Matrix</td></tr>
<tr><td style="width: 25%;"><a href="get_negconruv.html">get_negconeval</a></td>
<td>Get Negative and Positive Controls</td></tr>
<tr><td style="width: 25%;"><a href="get_negconruv.html">get_negconeval-method</a></td>
<td>Get Negative and Positive Controls</td></tr>
<tr><td style="width: 25%;"><a href="get_negconruv.html">get_negconruv</a></td>
<td>Get Negative and Positive Controls</td></tr>
<tr><td style="width: 25%;"><a href="get_negconruv.html">get_negconruv-method</a></td>
<td>Get Negative and Positive Controls</td></tr>
<tr><td style="width: 25%;"><a href="get_normalized.html">get_normalized</a></td>
<td>Retrieve Normalized Matrix</td></tr>
<tr><td style="width: 25%;"><a href="get_normalized.html">get_normalized-method</a></td>
<td>Retrieve Normalized Matrix</td></tr>
<tr><td style="width: 25%;"><a href="get_params.html">get_params</a></td>
<td>Extract scone parameters</td></tr>
<tr><td style="width: 25%;"><a href="get_params.html">get_params-method</a></td>
<td>Extract scone parameters</td></tr>
<tr><td style="width: 25%;"><a href="get_negconruv.html">get_poscon</a></td>
<td>Get Negative and Positive Controls</td></tr>
<tr><td style="width: 25%;"><a href="get_negconruv.html">get_poscon-method</a></td>
<td>Get Negative and Positive Controls</td></tr>
<tr><td style="width: 25%;"><a href="get_qc.html">get_qc</a></td>
<td>Get Quality Control Matrix</td></tr>
<tr><td style="width: 25%;"><a href="get_qc.html">get_qc-method</a></td>
<td>Get Quality Control Matrix</td></tr>
<tr><td style="width: 25%;"><a href="get_scores.html">get_scores</a></td>
<td>Extract scone scores</td></tr>
<tr><td style="width: 25%;"><a href="get_scores.html">get_scores-method</a></td>
<td>Extract scone scores</td></tr>
<tr><td style="width: 25%;"><a href="get_scores.html">get_score_ranks</a></td>
<td>Extract scone scores</td></tr>
<tr><td style="width: 25%;"><a href="get_scores.html">get_score_ranks-method</a></td>
<td>Extract scone scores</td></tr>
<tr><td style="width: 25%;"><a href="control_genes.html">housekeeping</a></td>
<td>Data: Positive and Negative Control Genes</td></tr>
<tr><td style="width: 25%;"><a href="control_genes.html">housekeeping_revised</a></td>
<td>Data: Positive and Negative Control Genes</td></tr>
<tr><td style="width: 25%;"><a href="impute_expectation.html">impute_expectation</a></td>
<td>Imputation of zero abundance based on general zero-inflated model</td></tr>
<tr><td style="width: 25%;"><a href="impute_null.html">impute_null</a></td>
<td>Null or no-op imputation</td></tr>
<tr><td style="width: 25%;"><a href="lm_adjust.html">lm_adjust</a></td>
<td>Linear Adjustment Normalization</td></tr>
<tr><td style="width: 25%;"><a href="make_design.html">make_design</a></td>
<td>Make a Design Matrix</td></tr>
<tr><td style="width: 25%;"><a href="metric_sample_filter.html">metric_sample_filter</a></td>
<td>Metric-based Sample Filtering: Function to filter single-cell RNA-Seq libraries.</td></tr>
<tr><td style="width: 25%;"><a href="scone.html">scone</a></td>
<td>Normalize Expression Data and Evaluate Normalization Performance</td></tr>
<tr><td style="width: 25%;"><a href="scone.html">scone-method</a></td>
<td>Normalize Expression Data and Evaluate Normalization Performance</td></tr>
<tr><td style="width: 25%;"><a href="SconeExperiment-class.html">SconeExperiment</a></td>
<td>Class SconeExperiment</td></tr>
<tr><td style="width: 25%;"><a href="SconeExperiment-class.html">SconeExperiment-class</a></td>
<td>Class SconeExperiment</td></tr>
<tr><td style="width: 25%;"><a href="SconeExperiment-class.html">SconeExperiment-method</a></td>
<td>Class SconeExperiment</td></tr>
<tr><td style="width: 25%;"><a href="sconeReport.html">sconeReport</a></td>
<td>SCONE Report Browser: Browse Evaluation of Normalization Performance</td></tr>
<tr><td style="width: 25%;"><a href="scone_easybake.html">scone_easybake</a></td>
<td>Wrapper for Running Essential SCONE Modules</td></tr>
<tr><td style="width: 25%;"><a href="score_matrix.html">score_matrix</a></td>
<td>SCONE Evaluation: Evaluate an Expression Matrix</td></tr>
<tr><td style="width: 25%;"><a href="SCRAN_FN.html">SCRAN_FN</a></td>
<td>Simple deconvolution normalization wrapper</td></tr>
<tr><td style="width: 25%;"><a href="select_methods.html">select_methods</a></td>
<td>Get a subset of normalizations from a SconeExperiment object</td></tr>
<tr><td style="width: 25%;"><a href="select_methods.html">select_methods-method</a></td>
<td>Get a subset of normalizations from a SconeExperiment object</td></tr>
<tr><td style="width: 25%;"><a href="simple_FNR_params.html">simple_FNR_params</a></td>
<td>Fit Simple False-Negative Model</td></tr>
<tr><td style="width: 25%;"><a href="subsample_cells.html">subsample_cells</a></td>
<td>Function to subsample Scone object by subsampling cells</td></tr>
<tr><td style="width: 25%;"><a href="subsample_cells_with_min_bio.html">subsample_cells_with_min_bio</a></td>
<td>Internal function to subsample Scone object by subsampling cells by size of smallest bio group</td></tr>
<tr><td style="width: 25%;"><a href="subsample_genes.html">subsample_genes</a></td>
<td>Function to subsample Scone object by subsampling genes</td></tr>
<tr><td style="width: 25%;"><a href="subsample_scone.html">subsample_scone</a></td>
<td>Wrapper Function to subsample Scone object</td></tr>
<tr><td style="width: 25%;"><a href="SUM_FN.html">SUM_FN</a></td>
<td>Sum scaling normalization function</td></tr>
<tr><td style="width: 25%;"><a href="TMM_FN.html">TMM_FN</a></td>
<td>Weighted trimmed mean of M-values (TMM) scaling normalization wrapper function</td></tr>
<tr><td style="width: 25%;"><a href="UQ_FN.html">UQ_FN</a></td>
<td>Upper-quartile (UQ) scaling normalization wrapper function</td></tr>
</table>
</body></html>
